recipes/lua/all/conanfile.py

```diff
@@ -46,17 +49,16 @@ def build(self):
 
     def package(self):
         # Extract the License/s from the header to a file
-        tmp = tools.load( os.path.join( self._source_subfolder, "src", "lua.h") )
+        tmp = tools.load( os.path.join(self.source_folder, self._source_subfolder, "src", "lua.h") )
```

Comment:
Why do you need from source_folder ? As all sources are copied to build_folder, you can use it from build_folder I think

Reply:
I added this because of the development workflow,
https://docs.conan.io/en/latest/developing_packages/package_dev_flow.html

 when you specify and extra source folder, the source is somewhere different than the conan file, and only having a relative path will not work

Reply:
Recipes everywhere generally don't support the development workflow. I don't think the change is bad, but it's very arbitrary and requires a more general solution to the problem.

Reply:
I agree with you @Minimonium , that we need a general solution, 
because I think for development, the way from the manual is good, and it should be fully supported.
it is basically an easy fix, build and package should start in the self.source_folder
because this is what is expected anyway and what is going on. This is why all the index center recipes work.
But I do not know where to suggest this...

Reply:
I also believe this `self.source_folder` is out of place here.

In the past, I was also using the development flow for developing packages.
I've converted to using  `conan create` by [caching the downloads](https://docs.conan.io/en/latest/configuration/download_cache.html) and using `-ks`.
I then make my modifications in `~/.conan/data/XXX/x.y.z/_/_/source`, which I track using `git` so I can eventually create patches.
@uilianries @Minimonium How are you developing your recipes?

Reply:
I don't have big recipe development cycles since my toolchains are simple. I previously tried to integrate the development flow into our CI, but quickly discovered insufficiencies with it and the lack of support in most existing recipes - so I ditched it for the `conan create`. 
I don't see a problem with the change here, but I don't think the dev flow can be fixed as simply as changing the default current path.

Reply:
the reason for the lack is, that conan expect that build and package you are in the self.source_folder, and that this is the same as the file where conanfile.py is.

as said, this is an easy fix for conan, just cd into the self.source_folder
no one will notice anything, except that things will start to work that did not work

want to verify?
just add a `with tools.chdir(self.source_folder):` into the build and the package method.
And all the sudden the development workflow, but also the usual conan create way, will work.

using / keeping things in .conan/data is super uncomfortable, long paths , crud names, ... getting everything in a VS Code workspace will just work with work arounds, ....
seriously, if this would have been the recommended way to go when I started with conan, I might have given up on it

anyhow, I am not sure if this is the right place to discuss this topic
and about the change, it just makes a relative path absolute , so what should be wrong with it?

Reply:
I finally raised an issue, that hopefully will lead to an advice how to handle the situation, please see https://github.com/conan-io/conan/issues/6629

Reply:
@a4z Use `conan create` approach for CCI recipes. Development flow and `conan create` are not totally compatible.

Reply:
removed now this change since it created to much controversial